src/array.jl

```diff
@@ -241,14 +324,24 @@ convert(::Type{CategoricalMatrix}, A::AbstractMatrix{T}) where {T} =
 convert(::Type{CategoricalMatrix{T}}, A::CategoricalMatrix{T}) where {T} = A
 convert(::Type{CategoricalMatrix}, A::CategoricalMatrix) = A
 
-function convert(::Type{CategoricalArray{T, N, R}}, A::AbstractArray{S, N}) where {S, T, N, R}
-    res = CategoricalArray{T, N, R}(undef, size(A))
+convert(::Type{CategoricalArray{T, N, R}}, A::AbstractArray{S, N}) where {S, T, N, R} =
+    _convert(CategoricalArray{T, N, R}, A)
+
+function _convert(::Type{CategoricalArray{T, N, R}}, A::AbstractArray{S, N};
```

Comment:
why `_convert` does not have `order` kwarg?

Reply:
That's just a matter of style. Since calling `ordered!` after the fact is as fast as calling it before filling the array, we do it in the caller after calling `_convert`. If you think it would be clearer to pass it as an argument, I can do that.

Reply:
Are there cases when you would want to avoid calling `ordered!` after calling `_convert`? If not (I think not), it would be clearer to have it inside `_convert` so that it does not get "forgotten" in some code path.

Reply:
Yes, the call from `convert` above doesn't set levels not order (as `convert` doesn't support any arguments).

Reply:
OK - then I think it is OK as is.